BUILD

```diff
@@ -91,28 +47,34 @@ gentbl_cc_library(
             ],
             "include/mlir-tcp/Dialect/IR/TcpDialect.cpp.inc",
         ),
+        (
+            ["-gen-attrdef-decls"],
+            "include/mlir-tcp/Dialect/IR/TcpAttrs.h.inc",
+        ),
+        (
+            ["-gen-attrdef-defs"],
+            "include/mlir-tcp/Dialect/IR/TcpAttrs.cpp.inc",
+        ),
+        (
+            ["-gen-enum-decls"],
+            "include/mlir-tcp/Dialect/IR/TcpEnums.h.inc",
+        ),
+        (
+            ["-gen-enum-defs"],
+            "include/mlir-tcp/Dialect/IR/TcpEnums.cpp.inc",
+        ),
     ],
     tblgen = "@llvm-project//mlir:mlir-tblgen",
     td_file = "include/mlir-tcp/Dialect/IR/TcpOps.td",
-    deps = [
-        ":TcpTdFiles",
-    ],
+    deps = [":TcpDialectTdFiles"],
 )
 
 cc_library(
     name = "TcpDialect",
-    srcs = [
-        "lib/Dialect/IR/TcpDialect.cpp",
-        "lib/Dialect/IR/TcpOps.cpp",
-    ],
-    hdrs = [
-        "include/mlir-tcp/Dialect/IR/TcpDialect.h",
-        "include/mlir-tcp/Dialect/IR/TcpOps.h",
-    ],
+    srcs = glob(["lib/Dialect/IR/*.cpp"]),
+    hdrs = glob(["include/mlir-tcp/Dialect/IR/*.h"]),
```

Comment:
Nit: can we avoid `glob`s here?  They tend to accidentally pull in unnecessary files.

Reply:
Sure, fIxed this and all other uses of `glob` in main BUILD.